### PR TITLE
Deferrable mode for CreateHyperparameterTuningJobOperator

### PR DESCRIPTION
**Pull request by moiseenkov**

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/vertex_ai/hyperparameter_tuning_job.py [hunk 1]
```diff
@@ -413,3 +422,108 @@ def delete_hyperparameter_tuning_job(
             metadata=metadata,
         )
         return result
+
+
+class HyperparameterTuningJobAsyncHook(GoogleBaseHook):
+    """Async hook for Google Cloud Vertex AI Hyperparameter Tuning Job APIs."""
+
+    # sync_hook_class = HyperparameterTuningJobHook
```
Comment (MaksYermak) [inline airflow/providers/google/cloud/hooks/vertex_ai/hyperparameter_tuning_job.py:430]: Do we need this comment?

File: tests/providers/google/cloud/triggers/test_vertexai.py
Comment (MaksYermak) [inline tests/providers/google/cloud/triggers/test_vertexai.py:1]: Could we update the filename to `test_vertex_ai.py` ?

File: airflow/providers/google/cloud/hooks/vertex_ai/hyperparameter_tuning_job.py (hunk 1 above)
Comment (moiseenkov) [reply airflow/providers/google/cloud/hooks/vertex_ai/hyperparameter_tuning_job.py]: Fixed, thanks for noticing

File: tests/providers/google/cloud/triggers/test_vertexai.py
Comment (moiseenkov) [reply tests/providers/google/cloud/triggers/test_vertexai.py:1]: Done

**Review by MaksYermak — APPROVED** [review]

**Pull request closed without merging**